### DISCUSSION TITLE
Orientations relationships using two ebsd files

### DISCUSSION THREAD

**Discussion by AGHz97** [ORTools4MTEX/ORTools#28] [Q&A]
Hello Frank, 
I need your help to analyze the orientation relationships between parent grains and child grains. I have **two** EBSD data files in .txt format such as the first one contains data of parent grains (ferritic grains) and the second one contains data of parent and child grains (more precisely austenoferritic grains). My question is: can I study the orientation relationships by defining the first file as the file that contains the orientations of the parent grains and the second one taking into account only the orientations of the child grains (austenitic grains)? I tried the `setParentGrainreconstructor` functions but we can only define one EBSD data file,
 thank you for your help

**Comment by frankNiessen** [comment]
Hi, 

I am not sure I fully understand what you are trying to do. To do a parent grain reconstruction you will need the child phase and parent phase within the same EBSD map, i.e., in the exact same area of your sample. If you just want to analyze the parent-child orientation relationship, I cannot see how the map only containing parent phase will be of help. You either need child-child-boundaries or child-parent boundaries to be able to measure the orientation relationship. If you have the parent and child phases in different maps, there are no boundaries of these phases that can be analyzed. Could you maybe try to tell me in more exact words what you want to achieve?

**Reply by AGHz97** [reply]
Hello Frank, 
I apologize for not being more specific in my question. To provide more details,  I am actually interested in creating 3D virtual microstructures of austenoferritic steels using a code that can transform from a parent ferritic grain microstructure to a synthetic microstructure with austenoferritic grains (75% austenite and 25% ferrite) as shown in the figure below:
![scheme1](https://user-images.githubusercontent.com/108796260/214444632-33336a8b-a350-425a-a903-1253258a6ee4.jpg)

Synthetic ferritic grains (which make up 25% of the phase proportion in the synthetic microstructure) maintain the same orientations as the parent ferritic grain. The goal is to validate this code for generating austenoferritic microstructures by comparing the orientation relationships between austenite and ferrite with those of Kurjumov-Sachs. Initially, I wanted to go directly through synthetic ferritic grains, but since they only make up 25% of the phase, it is difficult to reconstruct them, and I did not get good results. However, I thought of using the orientations of the parent ferritic grains (from the initial microstructure) because they have the same orientations as the synthetic ferritic grains. 
So my question is: Can we define the orientations of the parent grains using the microstructure data of the parent grains (initial microstructure) and analyze  it with the orientations of synthetic austenitic grains (child grains)? That's why I asked you my previous question: Can we define parent grains and child grains using two different files? 
Thank you again.

**Reply by frankNiessen** [reply]
Hi, 

I think I am slowly starting to understand where you are going :). So my initial idea would be to reconstruct the parent phase in your austenoferritic microstructure, I think this should be possible given the right reconstruction technique. Have you tried a simple growth algorithm? After having defined the `job` object and having optimized or set the orientation relationship you could try to grow the parent phase into the child phase by applying 

```
for k = 1:3
    job.calcGBVotes('p2c','threshold',k*2.5*degree);
    job.calcParentFromVote
end
```

Let me know if this works, you would then have the prior parent orientation of each child orientation. 
Combining the data from two maps is at least no standard procedure. In the MATLAB, MTEX, ORTools world anything can be done, but it might just involve a lot of work from your side, there is no pre-built way as far as I am aware.

Best wishes
Frank
